src/bin/coverage.rs

```diff
@@ -203,16 +206,16 @@ fn process(file: walkdir::DirEntry, verbose: bool) -> ParseResult {
         let mut ambiguity_result = Ok(());
         let start = Instant::now();
         let status = match result {
-            Ok(handle) => {
+            Ok(Ok(handle)) => {
                 ambiguity_result = ambiguity_check(handle);
                 if ambiguity_result.is_ok() {
                     ParseResult::Unambiguous
                 } else {
                     ParseResult::Ambiguous
                 }
             }
-            Err(parse::ParseError::TooShort(_)) => ParseResult::Partial,
-            Err(parse::ParseError::NoParse) => ParseResult::Error,
+            Ok(Err(_)) => ParseResult::Partial,
+            Err(_) => ParseResult::Error,
```

Comment:
These should be distinguished better, `error.partial.is_some()` perhaps? But `ModuleContents` will parse an empty string, so you want to check if the `partial` handle's `range` is empty. Not sure how easy that is though.

Reply:
I think with the flattened result this is clearer now?